addons/udes_stock/models/stock_picking.py

```diff
@@ -393,3 +406,48 @@ def _create_move(self, move_values):
         """
         Move = self.env["stock.move"]
         return Move.create(move_values)
+
+    def find_empty_pickings_to_unlink(self):
+        """Finds empty pickings to unlink in self, when it is set, otherwise
+        searches for any empty picking.
+
+        Filter out of the generic search when the picking type does not
+        have auto unlink empty enabled.
+        """
+        PickingType = self.env["stock.picking.type"]
+
+        # All pickings to be unlinked should not be marked
+        domain = [("u_mark", "=", False)]
+        if self:
+            domain.append(("id", "in", self.ids))
+        else:
+            pts = PickingType.search([("u_auto_unlink_empty", "=", False)])
+            if pts:
+                domain.append(("picking_type_id", "not in", pts.ids))
+        while True:
+            records = self.search(domain, limit=1000)
+            if not records:
+                break
+            yield records
+
+    def unlink_empty(self):
```

Comment:
Same here, do we need this for push from drop?

Reply:
Think this is just needed when refactoring. I will remove it along with, find_empty_pickings_to_unlink(), and the u_mark field.